docs/api/migration-guides/qiskit-1.0.mdx

```diff
@@ -0,0 +1,594 @@
+The *code* of the core of Qiskit lived in `qiskit-terra`, which was the owner of the root of the Python package `qiskit` (in other words, it controlled what happened when you ran `import qiskit`).
+Until Qiskit 1.0, the `qiskit` package was a namespace package and contained a second namespace package within itself at `qiskit.providers`.
+
+This caused us and our users quite a few problems over the years.
+
+For example, downstream libraries that depended on Qiskit often only actually needed the compiler core, and did not require the rest of the large ecosystem that came with `pip install qiskit`.
+They would therefore correctly specify their requirement as being on `qiskit-terra`.
+When people attempted to uninstall Qiskit, they would run `pip uninstall qiskit`, but `pip` has major problems here:
+
+1. `pip` does not remove now-unused distributions, so `pip uninstall qiskit` used to do almost nothing; there was no code in the distribution, so no code was removed.
+2. Even if it *were* to remove code, many downstream distributions would remain installed because they depended on the inner `qiskit-terra`.
+3. Even if *this* was uninstalled, it might still leave an importable `qiskit` directory with no usable code, because it was a namespace package.
+
+When installing or upgrading distributions with a `pip install` command, `pip` also does not take into account previous requirement resolutions.
+As there were two packages, upgrading a package that required `qiskit-terra` to be upgraded caused an invalid environment; `pip` would upgrade `qiskit-terra` but leave `qiskit` untouched.
+It would print out a warning on this and all subsequent `pip install` commands, but since nothing appeared broken, this largely just trained users to ignore the warning, and `pip` would not raise an error status or forbid operations.
+
+Over time, we removed elements from the metapackage until only `qiskit-terra` remained, as components were retired, or drifted away from really being part of the "necessary" installation of Qiskit (which was always more than was strictly necessary anyway).
+Of these components, `qiskit-aer` still exists and is actively updated, it's just always installed as a separate distribution now.
+From Qiskit 0.44 onwards, the only component of the `qiskit` distribution has been `qiskit-terra`.
+
+Similarly, we ever more strongly discouraged other libraries from using the namespace hooks.
+We removed the last of the Qiskit-family use of the hooks in non-obsolete packages with the release of Qiskit Aer 0.11 and its new `qiskit_aer` Python package, although until Qiskit 1.0 we also did some internal magic to make the namespace path `qiskit.providers.aer` continue to work.
+Starting from Qiskit 1.0, we have completely removed the ability for packages to extend any `qiskit` namespace, meaning that `pip uninstall` on the correct distribution in a valid environment will work as expected.
+
+
```

Comment:
Oh, I didn't actually finish writing this section.  I'll add on a bit more later.

Reply:
Done in [a25a489](https://github.com/Qiskit/documentation/pull/710/commits/a25a489d235fde47e5643118722c7fcc621009ae).